Issue with postgres store

**Describe the problem**

A clear and concise description of what the problem is.

```log
management/server/http/middleware/auth_middleware.go:88: Error when validating JWT claims: extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters;
```

Believe this related to a Postgres limit:
https://klotzandrew.com/blog/postgres-passing-65535-parameter-limit/

**To Reproduce**

Unknown

**Expected behavior**

For 65535 limit not to be reached

**Are you using NetBird Cloud?**

No

**NetBird version**

0.27.10

Some more logs
```
2024-06-19T01:56:16Z DEBG management/server/sql_store.go:188: took 273 ms to persist an account to the store
2024-06-19T01:56:16Z ERRO management/server/http/middleware/auth_middleware.go:88: Error when validating JWT claims: extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters; extended protocol limited to 65535 parameters
2024-06-19T01:56:16Z ERRO management/server/http/util/util.go:80: got a handler error: token invalid
2024-06-19T01:56:16Z ERRO management/server/telemetry/http_api_metrics.go:181: HTTP response 2515859429: GET /api/users status 401
2024-06-19T01:56:16Z ERRO management/server/telemetry/http_api_metrics.go:181: HTTP response 2515859429: GET /api/users status 401
```

--- a/management/server/sql_store.go
+++ b/management/server/sql_store.go
@@ -662,11 +662,7 @@ func NewSqliteStore(ctx context.Context, dataDir string, metrics telemetry.AppMe
 	}
 
 	file := filepath.Join(dataDir, storeStr)
-	db, err := gorm.Open(sqlite.Open(file), &gorm.Config{
-		Logger:          logger.Default.LogMode(logger.Silent),
-		CreateBatchSize: 400,
-		PrepareStmt:     true,
-	})
+	db, err := gorm.Open(sqlite.Open(file), getGormConfig())
 	if err != nil {
 		return nil, err
 	}
@@ -676,15 +672,20 @@ func NewSqliteStore(ctx context.Context, dataDir string, metrics telemetry.AppMe
 
 // NewPostgresqlStore creates a new Postgres store.
 func NewPostgresqlStore(ctx context.Context, dsn string, metrics telemetry.AppMetrics) (*SqlStore, error) {
-	db, err := gorm.Open(postgres.Open(dsn), &gorm.Config{
-		Logger:      logger.Default.LogMode(logger.Silent),
-		PrepareStmt: true,
-	})
+	db, err := gorm.Open(postgres.Open(dsn), getGormConfig())
 	if err != nil {
 		return nil, err
 	}
 
 	return NewSqlStore(ctx, db, PostgresStoreEngine, metrics)
+}
+
+func getGormConfig() *gorm.Config {
+	return &gorm.Config{
+		Logger:          logger.Default.LogMode(logger.Silent),
+		CreateBatchSize: 400,
+		PrepareStmt:     true,
+	}
 }
 
 // newPostgresStore initializes a new Postgres store.

--- a/management/server/sql_store_test.go
+++ b/management/server/sql_store_test.go
@@ -41,7 +41,7 @@ func TestSqlite_NewStore(t *testing.T) {
 }
 
 func TestSqlite_SaveAccount_Large(t *testing.T) {
-	if runtime.GOOS != "linux" && os.Getenv("CI") == "true" {
+	if runtime.GOOS != "linux" && os.Getenv("CI") == "true" || runtime.GOOS == "windows" {
 		t.Skip("skip large test on non-linux OS due to environment restrictions")
 	}
 	t.Run("SQLite", func(t *testing.T) {

--- a/management/server/sql_store_test.go
+++ b/management/server/sql_store_test.go
@@ -41,13 +41,14 @@ func TestSqlite_NewStore(t *testing.T) {
 }
 
 func TestSqlite_SaveAccount_Large(t *testing.T) {
-	if runtime.GOOS == "windows" {
-		t.Skip("The SQLite store is not properly supported by Windows yet")
+	if runtime.GOOS != "linux" && os.Getenv("CI") == "true" {
+		t.Skip("skip large test on non-linux OS due to environment restrictions")
 	}
 	t.Run("SQLite", func(t *testing.T) {
 		store := newSqliteStore(t)
 		runLargeTest(t, store)
 	})
+	// create store outside to have a better time counter for the test
 	store := newPostgresqlStore(t)
 	t.Run("PostgreSQL", func(t *testing.T) {
 		runLargeTest(t, store)

--- a/management/server/sql_store_test.go
+++ b/management/server/sql_store_test.go
@@ -44,8 +44,18 @@ func TestSqlite_SaveAccount_Large(t *testing.T) {
 	if runtime.GOOS == "windows" {
 		t.Skip("The SQLite store is not properly supported by Windows yet")
 	}
+	t.Run("SQLite", func(t *testing.T) {
+		store := newSqliteStore(t)
+		runLargeTest(t, store)
+	})
+	store := newPostgresqlStore(t)
+	t.Run("PostgreSQL", func(t *testing.T) {
+		runLargeTest(t, store)
+	})
+}
 
-	store := newSqliteStore(t)
+func runLargeTest(t *testing.T, store Store) {
+	t.Helper()
 
 	account := newAccountWithId(context.Background(), "account_id", "testuser", "")
 	groupALL, err := account.GetGroupAll()
@@ -54,7 +64,7 @@ func TestSqlite_SaveAccount_Large(t *testing.T) {
 	}
 	setupKey := GenerateDefaultSetupKey()
 	account.SetupKeys[setupKey.Key] = setupKey
-	const numPerAccount = 2000
+	const numPerAccount = 6000
 	for n := 0; n < numPerAccount; n++ {
 		netIP := randomIPv4()
 		peerID := fmt.Sprintf("%s-peer-%d", account.Id, n)

